modules/tinymce/src/core/main/ts/html/InvalidNodes.ts

```diff
@@ -159,17 +159,31 @@ const hasClosest = (node: AstNode, parentName: string): boolean => {
   return false;
 };
 
-const isInvalid = (schema: Schema, node: AstNode, parent: AstNode | null | undefined = node.parent): boolean => {
+const isInvalid = (schema: Schema, node: AstNode, isInsert: boolean, parent: AstNode | null | undefined = node.parent): boolean => {
+  if (!parent) {
+    return false;
+  }
+
   // Check if the node is a valid child of the parent node. If the child is
   // unknown we don't collect it since it's probably a custom element
-  if (parent && schema.children[node.name] && !schema.isValidChild(parent.name, node.name)) {
+  if (schema.children[node.name] && !schema.isValidChild(parent.name, node.name)) {
     return true;
+  }
+
   // Anchors are a special case and cannot be nested
-  } else if (parent && node.name === 'a' && hasClosest(parent, 'a')) {
+  if (node.name === 'a' && hasClosest(parent, 'a')) {
     return true;
-  } else {
-    return false;
   }
+
+  // heading element is valid if it is the only one child of summary (though on inserting it is still invalid)
+  if ([ 'h1', 'h2', 'h3', 'h4', 'h5', 'h6' ].includes(node.name) && hasClosest(node, 'summary')) {
```

Comment:
This is the same as the call in `DetailsElements.ts` I think it would be cleaner to make a `main/ts/html/AstType` module that has the `isHeading` function here and use it here and in the details elements.